Buddies/Activities/ActivityTableVC.swift

```diff
@@ -149,8 +149,17 @@ class ActivityTableVC: UITableViewController, FilterSearchBarDelegate {
         cell.locationLabel.text = activity.locationText
         let dateRange = DateInterval(start: activity.startTime.dateValue(),
                                      end: activity.endTime.dateValue())
+    
+        
+        let pixelsPerChar: CGFloat = 10.0
+        
+        let charsFitInCell = Int(cell.frame.width/pixelsPerChar)
+        
+        let locStrLength = max(15, charsFitInCell - activity.title.count)
+        
+        let locStr = dateRange.rangePhrase(relativeTo: Date(), tryShorteningIfLongerThan: locStrLength)
         
-        cell.dateLabel.text = dateRange.rangePhrase(relativeTo: Date()).capitalized
+        cell.dateLabel.text = locStr.capitalized
```

Comment:
Interesting. Is there any reason plain constraints wouldn't work to solve this problem?? Though I guess you need to know how long the string is

Reply:
It was kind of a preference I ran with.  I want to support at least 15 characters of a date string at a minimum, but when more space allows, I like the longer string.  That's why I change it based on the title/screen size.